### PR TITLE
JetBrains: create settings UI

### PR DESCRIPTION
**Pull request by vdavid**
Closes https://github.com/sourcegraph/sourcegraph/issues/36113
Closes https://github.com/sourcegraph/sourcegraph/issues/36076
Closes https://github.com/sourcegraph/sourcegraph/issues/36123 (a simple auth solution)

Caveats:
 - Both the auth flow and some of the settings need to be better documented. This will be a separate effort. The goal here was to create the UI itself so that the settings can be modified without touching XML files.
 - When one changes the settings after the popup has been opened, the settings won't update until the IDE is restarted. Solving this needs https://github.com/sourcegraph/sourcegraph/pull/34802. Once both of these PRs are merged, we can add the dynamic updates.
 - Remote URL replacements are handled just a plain text field for now. It's probably a rarely used feature. We'll document this part a bit better, and then this solution will likely be suitable.

## Test plan
- Watch the Loom: https://www.loom.com/share/7511c4cce8a54a40b2db7e597f5c81e2
- Check the code
- This change doesn't apply to the standalone version as there are no settings there.

## App preview:

- [Web](https://sg-web-dv-jetbrains-settings-ui.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-psgycmeegn.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/jetbrains/src/main/java/com/sourcegraph/config/SettingsComponent.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    private final JBTextField defaultBranchNameTextField;
+    private final JBTextField remoteUrlReplacementsTextField;
+    private final JBCheckBox globbingCheckBox;
+
+    public SettingsComponent() {
+        JBLabel sourcegraphUrlLabel = new JBLabel("Sourcegraph URL:");
+        sourcegraphUrlTextField = new JBTextField();
+        //noinspection DialogTitleCapitalization
+        sourcegraphUrlTextField.getEmptyText().setText("https://sourcegraph.example.com");
+        sourcegraphUrlTextField.setToolTipText("If your company has your own Sourcegraph instance, set its URL here.\nThe default is “https://sourcegraph.com”.");
```
Comment (philipp-spiess) [inline client/jetbrains/src/main/java/com/sourcegraph/config/SettingsComponent.java:28]: We had issues with `“` on Windows before IIRC. I can try if it still compiles later if you want!

File: client/jetbrains/src/main/java/com/sourcegraph/config/SourcegraphSettingsState.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * these persistent application settings are stored.
+ */
+@State(
+    name = "com.sourcegraph.config.SourcegraphSettingsState",
+    storages = @Storage("SdkSettingsPlugin.xml")
+)
+public class SourcegraphSettingsState implements PersistentStateComponent<SourcegraphSettingsState> {
+
+    public String userId = "John Q. Public";
+    public boolean ideaStatus = false;
```
Comment (philipp-spiess) [inline client/jetbrains/src/main/java/com/sourcegraph/config/SourcegraphSettingsState.java:23]: Huh what are these fields about?

Comment (philipp-spiess) [reply client/jetbrains/src/main/java/com/sourcegraph/config/SourcegraphSettingsState.java:23]: I don't think this file is actually used and needed, so perhaps this was from an example that you copied? For us the persistance happens via [`client/jetbrains/src/main/java/com/sourcegraph/config/SourcegraphConfig.java`](https://github.com/sourcegraph/sourcegraph/pull/36478/files#diff-66d65c66fda48ebbe11a0b730ceeedfe6fa90bfa188cb53983fe57171f6eb3ba)` right?

**Review by philipp-spiess — APPROVED** [review]
Carlton Dance

File: client/jetbrains/src/main/java/com/sourcegraph/config/SettingsComponent.java (hunk 1 above)
Comment (vdavid) [reply client/jetbrains/src/main/java/com/sourcegraph/config/SettingsComponent.java]: Replaced them, thanks!
Also, Jesus, Java, it's 2022!

File: client/jetbrains/src/main/java/com/sourcegraph/config/SourcegraphSettingsState.java (hunk 2 above)
Comment (vdavid) [reply client/jetbrains/src/main/java/com/sourcegraph/config/SourcegraphSettingsState.java]: Yeah, it was the example. Deleted it! Thanks! :)

**Pull request merged**